### PR TITLE
Flexibleidealist/native team view component

### PR DESCRIPTION
**Pull request by flexibleidealist**
- Updates SingleProject screen and creates TeamView and OwnerView components
- TeamView and OwnerView components conditionally rendered on SingleProject screen depending on current user in Redux store
- updates getOneProject method in projects controller to populate owner property with user object instead of just the ID; this change needs further discussion because it affects how other screens/components will access owner information

### CODE REVIEW

File: backend/seed/data.js [hunk 1]
```diff
@@ -70,12 +70,42 @@ const insertData = async () => {
   });
   await user4.save();
 
+  const user5 = new User({
+    about: "I'm an engineer and I like it",
+    email: "engineer1@mail.com",
+    first_name: "Dude",
```
Comment (gooliajulia) [inline backend/seed/data.js:76]: Welcome Dude! 😎

File: backend/seed/data.js [hunk 2]
```diff
@@ -70,12 +70,42 @@ const insertData = async () => {
...
+    portfolio_projects: [],
+    rejected_projects: [],
+    role: "Software Engineer",
+  });
+  await user5.save();
+
+  const user6 = new User({
+    about: "I like nerd stuff. A lot.",
+    email: "lady@mail.com",
+    first_name: "Maria",
```
Comment (gooliajulia) [inline backend/seed/data.js:91]: Welcome Maria! 🤓

File: native/components/OwnerView/OwnerView.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            </Pressable>
+          ))}
+          {project.interested_applicants.length > 3 ? (
+            <Pressable onPress={() => showApplicants(project._id)}>
+              <Text>{`See all ${project.interested_applicants.length} applicants`}</Text>
+            </Pressable>
+          ) : null}
+        </View>
+      ) : null}
+    </View>
```
Comment (gooliajulia) [inline native/components/OwnerView/OwnerView.jsx:52]: I don't know how you managed to make what could have easily been a very funky looking block of code end up SO good looking. I'm taking notes 📝 

I'm curious if we have a preference for ternary's with trailing nulls or guard operators for circumstances like this one? I usually tend to use a guard operator to eliminate the extra null, but I kind of like the readability of the ? here and you managed to keep the null's not looking too out of place.

File: native/screens/SingleProject.jsx [hunk 4]
```diff
@@ -74,18 +66,57 @@ export const SingleProject = ({ navigation, route }) => {
...
+      <Text
+        style={styles.heading}
+      >{`Looking for collaborators who can commit ${hours} hours per week.`}</Text>
+      {reduxUser._id === project.owner ? <OwnerView project={project} /> : null}
     </View>
   ) : (
     <Text>Loading...</Text>
   );
 };
+
```
Comment (gooliajulia) [inline native/screens/SingleProject.jsx:102]: Look at you styling away!! 👏 😉

File: native/screens/SingleProject.jsx [hunk 5]
```diff
@@ -74,18 +66,57 @@ export const SingleProject = ({ navigation, route }) => {
...
+      <Text style={styles.title}>{project.title}</Text>
+      {project.seeking ? (
+        <View>
+          <Text style={styles.heading}>Seeking</Text>
+          <Text>{`${rolesSought}`}</Text>
+        </View>
+      ) : null}
+      <Text style={styles.heading}>Description</Text>
+      <Text style={styles.paragraph}>{project.description}</Text>
+      {project.team_members?.some((member) => member === reduxUser._id) ||
```
Comment (gooliajulia) [inline native/screens/SingleProject.jsx:78]: this is my first time seeing the some method... cool!

**Review by gooliajulia — APPROVED** [review]
This looks great to me, too 👍

File: native/screens/SingleProject.jsx [hunk 6]
```diff
@@ -29,6 +25,18 @@ export const SingleProject = ({ navigation, route }) => {
...
 
+  let rolesSought;
+  if (project.designer_count && project.engineer_count) {
+    rolesSought = 'UX Designer, Software Engineer';
+  } else if (project.designer_count && !project.engineer_count) {
+    rolesSought = 'UX Designer';
+  } else if (!project.designer_count && project.engineer_count) {
+    rolesSought = 'Software Engineer';
+  } else {
+    rolesSought = null;
```
Comment (gooliajulia) [inline native/screens/SingleProject.jsx:36]: For some reason this is rendering as "Seeking null" when I tested this as a non-member/non-owner user.
Also when I log in as different users, the rolesSought eternally becomes equal to null?

Should rolesSought be handled as state local to the component so it's sure to update with new users?

**Comment by gooliajulia** [comment]
Was able to test out this branch on my machine and the team view, owner view, and neither views successfully rendered depending on user 👍 

Only bug I found was the seeking comment I mentioned above.

File: native/components/OwnerView/OwnerView.jsx (hunk 3 above)
Comment (flexibleidealist) [reply native/components/OwnerView/OwnerView.jsx:52]: I went ahead and changed (most of) the ternary statements with trailing `null` and replaced with the && operator; I think it reads a little more cleanly.

File: native/screens/SingleProject.jsx (hunk 6 above)
Comment (flexibleidealist) [reply native/screens/SingleProject.jsx:36]: Updated this to add a condition to check the `engineer_count` and `designer_count` properties of the project and not to render the seeking info if they are both set to `0` even if the `seeking` property is set to `true`.

**Pull request merged**